File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CommonDialogs/ColorDialog/Functionality/CustomColorsPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.CommonDialogs.ColorDialog.Functionality
{
    public partial class CustomColorsPage : UserControl
    {
        public CustomColorsPage()
        {
            InitializeComponent();
        }

        private void CustomColorsPage_Load(object sender, EventArgs e)
        {
            //Set array colors to CustomColors property of ColorDialog
            this.mobjDemoColorDialog.CustomColors = new int[] { 0x80FFFF, 0x8080FF };
        }

        private void mobjChangeBackColorButton_Click(object sender, EventArgs e)
        {
            //Set represented Label backgroud color a initial color for ColorDialog
            this.mobjDemoColorDialog.Color = this.mobjColorlLabel.BackColor;
            //Show ColorDialog
            this.mobjDemoColorDialog.ShowDialog();
        }

        private void mobjDemoColorDialog_Closed(object sender, EventArgs e)
        {
            //Check dialog result for closed ColorDialog
            if (this.mobjDemoColorDialog.DialogResult == DialogResult.OK)
            {
                //Set selct Color as background color of represented Label
                this.mobjColorlLabel.BackColor = this.mobjDemoColorDialog.Color;
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CommonDialogs/ColorDialog/Functionality/CustomColorsPage.vb
Namespace CompanionKit.Controls.CommonDialogs.ColorDialog.Functionality

	Public Class CustomColorsPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        Private Sub CustomColorsPage_Load(sender As Object, e As EventArgs) Handles Me.Load
            'Set array colors to CustomColors property of ColorDialog
            Me.mobjDemoColorDialog.CustomColors = New Integer() {&H80FFFF, &H8080FF}
        End Sub

        Private Sub mobjChangeBackColorButton_Click(sender As Object, e As EventArgs) Handles mobjChangeBackColorButton.Click
            'Set represented Label backgroud color a initial color for ColorDialog
            Me.mobjDemoColorDialog.Color = Me.mobjColorlLabel.BackColor
            'Show ColorDialog
            Me.mobjDemoColorDialog.ShowDialog()
        End Sub

        Private Sub mobjDemoColorDialog_Closed(sender As Object, e As EventArgs) Handles mobjDemoColorDialog.Closed
            'Check dialog result for closed ColorDialog
            If Me.mobjDemoColorDialog.DialogResult = DialogResult.OK Then
                'Set selct Color as background color of represented Label
                Me.mobjColorlLabel.BackColor = Me.mobjDemoColorDialog.Color
            End If
        End Sub
	End Class

End Namespace